ambry-store/src/main/java/com.github.ambry.store/BlobStoreCompactor.java

```diff
@@ -190,34 +198,46 @@ A single compaction job could be performed across multiple compaction cycles (if
     4. CLEANUP - Cleans up the old log segments and their index files
      */
 
+    logger.trace("Resuming compaction");
     try {
+      compactionInProgress.set(true);
       while (isActive && !compactionLog.getCompactionPhase().equals(CompactionLog.Phase.DONE)) {
         CompactionLog.Phase phase = compactionLog.getCompactionPhase();
+        logger.debug("Starting cycle to compact {} in {}", compactionLog.getCompactionDetails(), dataDir);
         switch (phase) {
           case PREPARE:
+            logger.debug("Compaction PREPARE started for {} in {}", compactionLog.getCompactionDetails(), dataDir);
             compactionLog.markCopyStart();
             // fall through to COPY
           case COPY:
+            logger.debug("Compaction COPY started for {} in {}", compactionLog.getCompactionDetails(), dataDir);
             copy();
             if (isActive) {
               compactionLog.markCommitStart();
+              logger.debug("Compaction COMMIT started for {} in {}", compactionLog.getCompactionDetails(), dataDir);
               commit(false);
               compactionLog.markCleanupStart();
+              logger.debug("Compaction CLEANUP started for {} in {}", compactionLog.getCompactionDetails(), dataDir);
               waitTillRefCountZero();
+              logger.debug("Wait for ref count to go to 0 has been finished {} in {}",
+                  compactionLog.getCompactionDetails(), dataDir);
               cleanup(false);
+              logger.debug("Completing for {} in {}", compactionLog.getCompactionDetails(), dataDir);
               compactionLog.markCycleComplete();
             }
             break;
           default:
             throw new IllegalStateException("Illegal compaction phase: " + phase);
         }
       }
+      compactionInProgress.set(false);
```

Comment:
Should this not be in the `finally`? Don't you want to set this to false in all exit paths from this method?